### PR TITLE
[RFC] Issue 5449 Added fix for modeline with number above range of MAX_INT

### PR DESCRIPTION
**Pull request by martsbradley**
I added code to check that the version number in the file was not longer than the number MAX_INT.

thanks
Martin

### CODE REVIEW

File: src/nvim/buffer.c [hunk 1]
```diff
@@ -4528,6 +4528,14 @@ chk_modeline (
           e = s + 4;
         else
           e = s + 3;
+
+        if (isVersionNumberTooLong(e))
+        {
+            return FAIL;
+        }
+
```
Comment (Shougo) [inline src/nvim/buffer.c]: Why it has three blank lines?
It should be one.

File: src/nvim/buffer.c [hunk 2]
```diff
@@ -4610,6 +4618,25 @@ chk_modeline (
 }
 
 /*
+ * checks the modeline is not longer
```
Comment (Shougo) [inline src/nvim/buffer.c]: Please use `//` style comments.

File: src/nvim/buffer.c [hunk 3]
```diff
@@ -4610,6 +4618,25 @@ chk_modeline (
...
+ * than the INT_MAX 
+ */
+static bool isVersionNumberTooLong(char_u* e)
+{
+  int digitsFound = 0;
+  char myBuffer[20];
+ 
+  snprintf(&myBuffer[0], 20, "%d",INT_MAX);
```
Comment (Shougo) [inline src/nvim/buffer.c]: It should be:

```c
snprintf(&myBuffer[0], sizeof(myBuffer), "%d",INT_MAX);
```

File: src/nvim/buffer.c (hunk 2 above)
Comment (phanimahesh) [inline src/nvim/buffer.c]: Do this only for newly added comments, and lines immediately surrounding any changes introduced. Run clint to get the list of changes required.

The codebase is littered with `/* ...*/` style comments and updating all of them at once will make git-blaming very difficult, hence this policy was chosen to update things incrementally.

File: src/nvim/buffer.c [hunk 4]
```diff
@@ -4610,6 +4618,25 @@ chk_modeline (
...
+  int longestVersionPossible = strlen(myBuffer);
+ 
+  for (digitsFound = 0;
+       ascii_isdigit(*e) && digitsFound < 20;
+       digitsFound++,e++);
```
Comment (phanimahesh) [inline src/nvim/buffer.c]: should this have an empty block `{}` to make it explicit? I find it easy to misread empty-body loops without explicit empty blocks.

**Comment by justinmk** [comment]
Thanks, but https://github.com/neovim/neovim/pull/5450/ (which is linked from #5449) covers this already.

**Pull request closed without merging**